.github/workflows/build_test.yaml

```diff
@@ -21,18 +23,31 @@ jobs:
 
       - name: Prepare
         run: |
-          sudo apt install lcov libsofthsm2 -y
+          sudo apt install lcov gcovr libsofthsm2 -y
 
-      - name: Build and test
-        run: |
-          mkdir build
-          cd build
-          cmake ../
-          cmake ../ -DWITH_TEST=ON -DWITH_COVERAGE=ON -DCMAKE_BUILD_TYPE=Debug
-          make; make tests; make coverage
+      - name: Install Build Wrapper
+        uses: SonarSource/sonarqube-scan-action/install-build-wrapper@v4
 
+      - name: Build using SonarQube Build Wrapper
+        run: |
+          cmake -S . -B build  -DCMAKE_BUILD_TYPE=Debug -DWITH_TEST=ON -DWITH_COVERAGE=ON -G "Unix Makefiles" -DCoverage_SONARQUBE=ON -DCMAKE_EXPORT_COMPILE_COMMANDS=ON
+          build-wrapper-linux-x86-64 --out-dir ${{ env.BUILD_WRAPPER_OUT_DIR }} cmake --build build/ --config Debug --parallel
+      - name: Run Tests
+        run: |          
+          (cd build && ctest --repeat until-pass:2 --output-on-failure && make coverage)
       - name: Upload codecov report
         uses: codecov/codecov-action@v4
         with:
           token: ${{ secrets.CODECOV_TOKEN }}
           files: ./build/coverage.total
+      - name: Prepare SonarQube coverage report
```

Comment:
Then we don't need "Upload codecov report". Which source is used in README.md for coverage? It should be the same.